### PR TITLE
[AAP-23818] Credentials plugins wizard test button

### PR DESCRIPTION
**Pull request by skochay330**
This PR adds 'test' button to the credentials plugins wizard

Jira issue: https://issues.redhat.com/browse/AAP-23818 

Screenshot 2024-05-21 at 5 50 18 PM

### CODE REVIEW

File: frontend/awx/access/credentials/CredentialForm.tsx [hunk 1]
```diff
@@ -578,7 +578,8 @@ function CredentialSubForm({
   setWatchedSubFormFields: (fields: unknown[]) => void;
 }) {
   const { t } = useTranslation();
-  const openCredentialPluginsModal = useCredentialPluginsModal();
+  const alertToaster = usePageAlertToaster();
```
Comment (kialam) [inline frontend/awx/access/credentials/CredentialForm.tsx:581]: I would move this directly into `CredentialPluginsModal` and avoid passing it in to the `useCredentialPluginsModal ` as a hook.

File: frontend/awx/access/credentials/CredentialForm.tsx [hunk 2]
```diff
@@ -578,7 +578,8 @@ function CredentialSubForm({
...
+  const openCredentialPluginsModal = useCredentialPluginsModal(alertToaster);
```
Comment (kialam) [inline frontend/awx/access/credentials/CredentialForm.tsx:582]:
```suggestion
  const openCredentialPluginsModal = useCredentialPluginsModal();
```

File: frontend/awx/access/credentials/CredentialPlugins/CredentialPlugins.tsx [hunk 3]
```diff
@@ -43,57 +51,59 @@ export function CredentialPlugins({
...
-                label={input.label}
-                labelHelp={input.help_text}
-                isRequired={credentialType.inputs.required.includes(input.id)}
-              />
-            );
-          }
-        })}
-      </PageFormSection>
-    );
+    } else
```
Comment (kialam) [inline frontend/awx/access/credentials/CredentialPlugins/CredentialPlugins.tsx:59]: This `else` is unnecessary.

File: frontend/awx/access/credentials/CredentialPlugins/hooks/useCredentialPluginsDialog.tsx [hunk 4]
```diff
@@ -1,10 +1,16 @@
 import { useTranslation } from 'react-i18next';
-import { PageFormSubmitHandler, usePageDialogs } from '../../../../../../framework';
-import { Modal, ModalVariant } from '@patternfly/react-core';
+import {
+  IPageAlertToaster,
+  PageFormSubmitHandler,
+  usePageDialogs,
+} from '../../../../../../framework';
```
Comment (kialam) [inline frontend/awx/access/credentials/CredentialPlugins/hooks/useCredentialPluginsDialog.tsx:6]:
```suggestion
import {
  IPageAlertToaster,
  PageFormSubmitHandler,
  usePageDialogs,
  usePageAlertToaster
} from '../../../../../../framework';
```

File: frontend/awx/access/credentials/CredentialPlugins/hooks/useCredentialPluginsDialog.tsx [hunk 5]
```diff
@@ -16,7 +22,9 @@ export interface CredentialPluginsModalProps {
   accumulatedPluginValues?: CredentialPluginsInputSource[];
 }
 
-function CredentialPluginsModal(props: CredentialPluginsModalProps) {
+function CredentialPluginsModal(
+  props: CredentialPluginsModalProps & { alertToaster: IPageAlertToaster }
+) {
```
Comment (kialam) [inline frontend/awx/access/credentials/CredentialPlugins/hooks/useCredentialPluginsDialog.tsx:27]:
```suggestion
function CredentialPluginsModal(
  props: CredentialPluginsModalProps
) {
```

File: frontend/awx/access/credentials/CredentialPlugins/hooks/useCredentialPluginsDialog.tsx [hunk 6]
```diff
@@ -16,7 +22,9 @@ export interface CredentialPluginsModalProps {
...
   const { t } = useTranslation();
```
Comment (kialam) [inline frontend/awx/access/credentials/CredentialPlugins/hooks/useCredentialPluginsDialog.tsx:28]:
```suggestion
  const { t } = useTranslation();
  const alertToaster = usePageAlertToaster();
```

File: frontend/awx/access/credentials/CredentialPlugins/hooks/useCredentialPluginsDialog.tsx [hunk 7]
```diff
@@ -57,6 +65,39 @@ function CredentialPluginsModal(props: CredentialPluginsModalProps) {
...
+        timeout: 2000,
+      };
+
+      const payload = {
+        metadata: rest,
+      };
+
+      postRequest(awxAPI`/credentials/${String(source_credential)}/test/`, payload)
+        .then(() => {
+          props.alertToaster.addAlert(alert);
```
Comment (kialam) [inline frontend/awx/access/credentials/CredentialPlugins/hooks/useCredentialPluginsDialog.tsx:84]:
```suggestion
          alertToaster.addAlert(alert);
```

File: frontend/awx/access/credentials/CredentialPlugins/hooks/useCredentialPluginsDialog.tsx [hunk 8]
```diff
@@ -57,6 +65,39 @@ function CredentialPluginsModal(props: CredentialPluginsModalProps) {
...
+        })
+        .catch((error) => {
+          props.alertToaster.addAlert({
```
Comment (kialam) [inline frontend/awx/access/credentials/CredentialPlugins/hooks/useCredentialPluginsDialog.tsx:87]:
```suggestion
          alertToaster.addAlert({
```

File: frontend/awx/access/credentials/CredentialPlugins/hooks/useCredentialPluginsDialog.tsx [hunk 9]
```diff
@@ -67,21 +108,24 @@ function CredentialPluginsModal(props: CredentialPluginsModalProps) {
...
         handleSubmit={handleSubmit}
+        handleTest={handleTest}
         defaultValues={getDefaultValues()}
       />
     </Modal>
   );
 }
 
-export function useCredentialPluginsModal() {
+export function useCredentialPluginsModal(alertToaster: IPageAlertToaster) {
```
Comment (kialam) [inline frontend/awx/access/credentials/CredentialPlugins/hooks/useCredentialPluginsDialog.tsx:118]:
```suggestion
export function useCredentialPluginsModal() {
```

File: frontend/awx/access/credentials/CredentialPlugins/hooks/useCredentialPluginsDialog.tsx [hunk 10]
```diff
@@ -67,21 +108,24 @@ function CredentialPluginsModal(props: CredentialPluginsModalProps) {
...
   const { pushDialog, popDialog } = usePageDialogs();
   const [props, setProps] = useState<CredentialPluginsModalProps>();
   useEffect(() => {
     if (props) {
-      pushDialog(<CredentialPluginsModal {...props} onClose={popDialog} />);
+      pushDialog(
+        <CredentialPluginsModal {...props} onClose={popDialog} alertToaster={alertToaster} />
```
Comment (kialam) [inline frontend/awx/access/credentials/CredentialPlugins/hooks/useCredentialPluginsDialog.tsx:124]:
```suggestion
        <CredentialPluginsModal {...props} onClose={popDialog} />
```

File: frontend/awx/access/credentials/CredentialPlugins/hooks/useCredentialPluginsDialog.tsx [hunk 11]
```diff
@@ -67,21 +108,24 @@ function CredentialPluginsModal(props: CredentialPluginsModalProps) {
...
+      );
     } else {
       popDialog();
     }
-  }, [props, pushDialog, popDialog]);
+  }, [props, pushDialog, popDialog, alertToaster]);
```
Comment (kialam) [inline frontend/awx/access/credentials/CredentialPlugins/hooks/useCredentialPluginsDialog.tsx:129]:
```suggestion
  }, [props, pushDialog, popDialog]);
```

**Review by kialam — CHANGES_REQUESTED** [review]
Looks good so far. I would move the toast alert into the credentials plugin modal rather than passing it down and remove the unnecessary `else` statement.

File: frontend/awx/access/credentials/CredentialPlugins/CredentialPlugins.tsx (hunk 3 above)
Comment (skochay330) [reply frontend/awx/access/credentials/CredentialPlugins/CredentialPlugins.tsx]: Leftover from debugging, good catch!

**Review by kialam — APPROVED** [review]
Should be good to go once the alert toaster is moved into their respective useCredentialPluginsModal and useCredentialTestModal hooks. ⭐

**Pull request merged**